src/NHibernate/Hql/Ast/ANTLR/QueryTranslatorImpl.cs

```diff
@@ -55,7 +55,9 @@ public partial class QueryTranslatorImpl : IFilterTranslator
 			_stageOneAst = parsedQuery;
 			_compiled = false;
 			_shallowQuery = false;
-			_enabledFilters = enabledFilters;
+			// Copy the filters, because a hack into EnterRule_insertStatement may clear them. It
+			// should be done only for the current query translator, not for the whole session.
+			_enabledFilters = new Dictionary<string, IFilter>(enabledFilters);
```

Comment:
The hack moreover requires a dictionary duplication, in order to avoid wrecking the session enabled filters.